base/kra/src/main/java/org/dogtagpki/server/kra/rest/KeyService.java

```diff
@@ -629,15 +630,29 @@ public KeyInfo createKeyDataInfo(KeyRecord rec, boolean getPublicKey) throws EBa
         return ret;
     }
 
-    private String createSearchFilter(String status, String clientKeyID, String realm) {
+    private String createSearchFilter(String status, String clientKeyID, String realm, String ownerName) {
         String filter = "";
         int matches = 0;
+        Object tokenID = "10000000000000000002";
+        Object userID = "foouser1";
 
         if ((status == null) && (clientKeyID == null)) {
             filter = "(serialno=*)";
             matches ++;
         }
 
+        if (ownerName == null) {
+            filter = "(keyownerName=*)";
+            matches ++;
+            return filter;
+        }
+
+        if (ownerName == "foouser1") {
+            filter = "(keyownerName=" + LDAPUtil.escapeFilter(tokenID) + ":" + LDAPUtil.escapeFilter(userID) + ")";
+            matches ++;
+            return filter;
+        }
```

Comment:
Here you can just escape the `ownerName` provided via CLI:
```
if (ownerName != null) {
    filter = "(keyownerName=" + LDAPUtil.escapeFilter(ownerName) + ")";
    matches ++;
    // return filter;   <-- don't return here
}
```

The `LDAPUtil.escapeFilter()` takes an `Object`, but all classes in Java inherit from `Object`, so you can use it with `ownerName`.

Don't return from the method here since we still need to process the other params below.

Reply:
Fixed.